### PR TITLE
refactor: move to modules, ox_lib, style fixes

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

Adds the modules, replaces the deprecated export and functions, switch commands and progress bars to lib, and then some minor style fixes

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Review by Manason — APPROVED** [review]

File: client.lua [hunk 1]
```diff
@@ -332,28 +337,27 @@ RegisterNetEvent('iens:repair', function()
 		QBCore.Functions.Notify(Lang:t("error.inside_veh_req"))
 		return
 	end
-	vehicle = cache.vehicle
 	if isNearMechanic() then return end
-	if GetVehicleEngineHealth(vehicle) >= cfg.cascadingFailureThreshold + 5 then
+	if GetVehicleEngineHealth(cache.vehicle) >= cfg.cascadingFailureThreshold + 5 then
 		QBCore.Functions.Notify(Lang:t(('nofix_message_%s'):format(noFixMessagePos)))
-		noFixMessagePos += 1
+		noFixMessagePos = noFixMessagePos + 1
```
Comment (BerkieBb) [inline client.lua]: This change was not necessary

File: client.lua [hunk 2]
```diff
@@ -332,28 +337,27 @@ RegisterNetEvent('iens:repair', function()
...
+	SetVehicleUndriveable(cache.vehicle, false)
+	SetVehicleEngineHealth(cache.vehicle, cfg.cascadingFailureThreshold + 5)
+	SetVehiclePetrolTankHealth(cache.vehicle, 750.0)
+	healthEngineLast = cfg.cascadingFailureThreshold + 5
+	healthPetrolTankLast = 750.0
+	SetVehicleEngineOn(cache.vehicle, true, false )
+	SetVehicleOilLevel(cache.vehicle,(GetVehicleOilLevel(cache.vehicle) / 3) - 0.5)
 	QBCore.Functions.Notify(Lang:t(('fix_message_%s'):format(fixMessagePos)))
-	fixMessagePos += 1
+	fixMessagePos = fixMessagePos + 1
```
Comment (BerkieBb) [inline client.lua]: Not necessary

File: fxmanifest.lua [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 client_script 'client.lua'
 server_script 'server.lua'
 
-lua54 'yes'
+modules {
+	'qbx-core:core',
+    'qbx-core:utils'
+}
+
+lua54 'yes'
```
Comment (BerkieBb) [inline fxmanifest.lua]: Add `use_experimental_fxv2_oal 'yes'` here to use oal, it lazy loads natives and in turn speeds up the resource (or well, lower its usage)

**Review by BerkieBb — APPROVED** [review]

**Comment by Manason** [comment]
Fix lint check failure

**Review by Manason — CHANGES_REQUESTED** [review]
Fix lint check failure

File: client.lua [hunk 4]
```diff
@@ -473,7 +479,6 @@ end
 CreateThread(function()
 	while true do
 		Wait(50)
-		local ped = cache.ped
 		if isPedDrivingAVehicle() then
```
Comment (Manason) [inline client.lua]:
```suggestion
		if cache.seat == -1 then
```

**Review by Manason — APPROVED** [review]

**Pull request merged**